### ISSUE TITLE
Creators update breaks RDWrap

### ISSUE THREAD

**Issue by Recka50** [stascorp/rdpwrap#229]
Updated to Windows 10 creators update and RDWrap is now broken (won't connect, I can't allow connections)

Tried re-installing RDWrap

**Comment by binarymaster** [comment]
This tells nothing. NT version? Edition?

**Comment by foxx1337** [comment]
Just to chime in here, I have the same problem.

This is RDPConf:
![RDPConf on Creators Update](https://i.imgur.com/8UFbIgS.png)

Windows 10 Professional x64, Version 1703, OS Build 15063.13
This is the final version, which I installed through the [Microsoft Setup Assistant](https://www.microsoft.com/en-us/software-download/windows10).

**Comment by rykp** [comment]
Same here. Instead, it shows it is not listening. Mine is Windows 10 Home X64. I had rdpwap installed and working fine before upgrade. I uninstalled rdpwrap. Upgraded to Creator via Windows 10 Upgrade Assistant. Downloaded Rdpwap 1.6.1 again and ran install.bat.

![untitled](https://cloud.githubusercontent.com/assets/26998849/24789372/8ff692e8-1b27-11e7-9512-8c6c8afe3534.png)

**Comment by binarymaster** [comment]
@foxx1337 your version is not up to date.

@rykp your issue is #194.

I'm waiting for the answer from @Recka50.

**Comment by rykp** [comment]
Thanks for your quick note. I see. Bad luck :( Have been using rdpwrap and love it. Look forward to a fix.

**Comment by sadmam** [comment]
Hey all, 

I came across this issue because after the creator update my install also got broken...

Tried  https://github.com/stascorp/rdpwrap/issues/45#issuecomment-182632817 and https://github.com/stascorp/rdpwrap/issues/194 but still a no go.

![capturar](https://cloud.githubusercontent.com/assets/17482168/24793591/8bfe0254-1b7a-11e7-97d1-3c88bb3a1423.PNG)

This is a print of the info that I get...
Os version 1703 and compilation 15063.13

Thanks for the help.

**Comment by binarymaster** [comment]
If your edition is Windows 10 Home and your termsrv build is 15004 and higher, then it WILL NOT work, because of #194 issue.

**Comment by sadmam** [comment]
@binarymaster Is anything I can do to help fix it?
Just a side note, it was working just fine before the creators update (witch i did yesterday... dumb self)... 
I know that they introduced something, and the size of the builds is different on > 10.0.15002.1001 builds, but #https://github.com/stascorp/rdpwrap/issues/194 was created on 14 Jan, where these people on the insiders program or something? How did they get > 10.0.15002.1001 this version so soon?

**Comment by binarymaster** [comment]
Yes, this problem was known before "Creators update", they are insiders indeed.

**Comment by binarymaster** [comment]
So, closing as duplicate of #194.

**Comment by JJMadrigal** [comment]
Hello. 

Any solution for this issue.

I have just installed the creators update and It does not work

Thnx

**Comment by MikeSam06** [comment]
hope master can give the solution instead of close and mention duplicate :(

**Comment by mrpentium** [comment]
Looks like the Administrator isn't listening either (lol).  Haven't heard a response

**Comment by markyi370** [comment]
@mrpentium @MikeSam06 @JJMadrigal It doesn't help for binarymaster to talk to everyone on every issue. So, he consolidates duplicate issues into one and performs discussion on one channel, making it alot easier for everyone. Complaining that he closed your issue when there is an identical one won't help. See #194, there is discussion happening there. That is the main issue you are all experiencing (well, most of you).

**Comment by explorerhk** [comment]
Just download and copy the following file to c:\windows\system32, then reboot, hope this helps.

64-bit Windows 10: https://github.com/stascorp/rdpwrap/files/1236856/rfxvmt.zip

32-bit Windows 10: https://github.com/stascorp/rdpwrap/files/1238499/rfxvmt.zip

**Comment by Sp0oner** [comment]
Thank you!!!!! That just fixed my issue with it not working and i've been waiting since updating to the creators update when it was released. When was that fix announced as I haven't seen it until now?!

**Comment by seaskyways** [comment]
Wow man this thing works . I started to explore other options just before i saw this ! Thanks man

**Comment by dvdp** [comment]
Works great for me too - a big thank you !!!

**Comment by andr0med** [comment]
Thank you, thank you, thank you!!!!  - did fresh install of rdp wrapper and added .dll file to system32 and now everything works.

**Comment by jplcampbell** [comment]
Sweet. werked for me tooo!

**Comment by Steve-1029** [comment]
Thank you Explorhk
Worked like a charm
Can you briefly describe why that works ?
I wonder why that fix isn't in the "UPDATE.bat" process ?
thanks again

**Comment by markyi370** [comment]
@Steve-1029 This file became necessary to add in the Creators Update. @binarymaster hasn't fixed update.bat yet to do so, and it would be illegal for us to distribute this file automatically.

**Comment by markyi370** [comment]
I can upload a 64-bit rfxvmt.dll from Windows 10 Pro 1709 Monday evening. Do I get it from system32 or SysWOW64?

**Comment by explorerhk** [comment]
Just in case if you guys have upgraded to Windows 10 Pro 1709, please find the newest file attached.

For x64

[rfxvmt_x64_Build1709.zip](https://github.com/stascorp/rdpwrap/files/1438515/rfxvmt_x64_Build1709.zip)

**Comment by Steve-1029** [comment]
Question for clarification:
the FIX for version 1703 build 15063  was your dll  named rfxvmt.zip
works great.  Working on several machines

Now Microsoft has a "Targeted Version" 1709 build 16299.19
but the Update tool does not reflect that build yet.  My test Win-Home machine says it is up to date on 1703.  I guess "Targeted" is similar to Beta test version ?

So I guess sometime in the future MSFT will force that update on me
and when that happens the RDP will be broken again
and then I need to use your new rfxvmt_64_Build1709.zip 

Sorry to be a little dense here but I am a Linux/MAC OSX user and I have been dragged kicking and screaming into the Win-10 world by one app that requires Windows

**Comment by dvdp** [comment]
Pro ? I guess that should be 'Home'. Afaik 'Pro' has RDP available so 
no need for RDPWrap ... 



Van: "explorerhk" <notifications@github.com> 
Aan: "stascorp/rdpwrap" <rdpwrap@noreply.github.com> 
Cc: "danny vandepol" <danny.vandepol@telenet.be>, "Comment" <comment@noreply.github.com> 
Verzonden: Donderdag 2 november 2017 17:19:50 
Onderwerp: Re: [stascorp/rdpwrap] Creators update breaks RDWrap (#229) 



Just in case if you guys have upgraded to Windows 10 Pro 1709, please find the newest file attached. 

For x64 

[ https://github.com/stascorp/rdpwrap/files/1438515/rfxvmt_x64_Build1709.zip | rfxvmt_x64_Build1709.zip ]

**Comment by explorerhk** [comment]
I just tested on both fresh install and upgrade version of Windows 10 Home 1709, it works fine.

1. Download RDPWrap v1.6
https://github.com/stascorp/rdpwrap/releases/download/v1.6.1/RDPWrap-v1.6.1.zip
2. Extract and install it.
3. Copy rfxvmt.dll to c:\windows\system32
4. Reboot

x64
https://github.com/stascorp/rdpwrap/files/1438515/rfxvmt_x64_Build1709.zip

x86
https://github.com/stascorp/rdpwrap/files/1443741/rfxvmt_x86_Build1709.zip

Hope it helps.

**Comment by sskras** [comment]
explorerhk wrote:
>
> Copy rfxvmt.dll to c:\windows\system32
> Reboot
>
> x64
> https://github.com/stascorp/rdpwrap/files/1438515/rfxvmt_x64_Build1709.zip
>
> x86
> https://github.com/stascorp/rdpwrap/files/1443741/rfxvmt_x86_Build1709.zip

It would be nice to have a way to get rfxvmt.dll from the original sources, though, as they can request the removal of these attachments from GitHub.  I am not sure about the latest version of the .dll file, but at least it is contained in these updates:

- Cumulative update for Windows 10 Version 1511: February 9, 2016
https://support.microsoft.com/en-us/help/3135173/cumulative-update-for-windows-10-version-1511-february-9--2016

- Cumulative update for Windows 10: February 9, 2016
https://support.microsoft.com/en-us/help/3135174/cumulative-update-for-windows-10-february-9--2016

The needed files are called this, I guess:

* windows10.0-kb3135173-x86.msu 
  (SHA256) D212BC27E7623935EACE95235678053A36AEE44EE74AA73A56D193567E0C8890
* windows10.0-kb3135173-x64.msu 
  (SHA256) E3474E1F4FF12C4BC3C7BD77760ACD2472C7D87E964AD7E1C3ECE73A089D2D0C
* windows10.0-kb3135174-x86.msu 
  (SHA256) 10420DB8A3D7AD7C1B4195F89828129CA89DF2B94886855BEEC171DD520EC774
* windows10.0-kb3135174-x64.msu 
  (SHA256) C179CF48BF9153A8390A789C1B1904CFE595AA5DFBA8E1E73F592E58AD572DDC

Random page that provides direct download URLs:
https://www.tenforums.com/windows-10-news/40282-cumulative-update-windows-10-version-1511-kb3135173.html

The other page tells download sizes:
http://ms-vnext.net/UpdateArchive/

http://download.windowsupdate.com/d/msdownload/update/software/secu/2016/02/windows10.0-kb3135173-x86_31f87a03e38211ca3572506c466c9e2eb0fcdb8e.msu (286.56 MB)
http://download.windowsupdate.com/c/msdownload/update/software/secu/2016/02/windows10.0-kb3135173-x64_8709ec87df2757c6077d2e7178dbf083473f138a.msu (612.44 MB)
http://download.windowsupdate.com/d/msdownload/update/software/secu/2016/02/windows10.0-kb3135174-x86_9c8184592c519d2bd1ada43d09f51e9eb01413f3.msu (270.34 MB)
http://download.windowsupdate.com/d/msdownload/update/software/secu/2016/02/windows10.0-kb3135174-x64_3c0cd8e1a7ad32c995c3ed492f20548a1e1cc4f7.msu (489.30 MB)

Maybe there are more recent version released, but I haven't found any newer for w10.

The files are quite large, and to get the needed "rfxvmt.dll" from them, the software will need to extract internal .cab file at first (using eg. extract.exe, builtin utility) and then extract the .dll from the .cab (using the same extract.exe).

This would be tedious but officially legal way to get the missing file (if the original is lost).

Maybe some older versions can be extracted from the installation files saved by Windows setup program to C: , but I have no idea about it's location since XP and the i386 directory.  Probably something useful could be extracted from SxS dir:
https://blogs.technet.microsoft.com/askcore/2008/09/17/what-is-the-winsxs-directory-in-windows-2008-and-windows-vista-and-why-is-it-so-large/

Any more elegant ideas on restoring the .dll ?

**Comment by mitchmenghi** [comment]
Sorry this does not work or resolve the LISTENER STATE issue in Win Pro 10 1709. I already tried it and its not working. Please advise.

Note: I only replaced rfxvmt.dll in \System32 not in \SYSWOW

**Comment by explorerhk** [comment]
Have you installed RDWrap before replacing the dll file? You also need to reboot your PC to take the changes

**Comment by mitchmenghi** [comment]
Yes i did, and i also rebooted... 100% doesnt work.... other versions have worked....

**Comment by binarymaster** [comment]
@mitchmenghi check the correctness of the version you copied to system32, maybe wrong architecture. Then try to **reinstall** RDP Wrapper.

**Comment by mabacon** [comment]
It was working for me on 1703, but not since 1709 update, since that update I can not get it working.
I have tried all the list of steps here, but it is fully supported but still not listening.
Tried: New x64 DLL, uninstall, install, update, reboot, checked firewall ports, disabled firewall.

Using Win 10 Home x64

**Comment by mer30hamid** [comment]
Thanks @explorerhk, Copying rfxvmt.dll file into Windows\System32 worked for me :)

Using Win 10 Home x86 (1709)

**Comment by Steve-1029** [comment]
I have had one success
RDP Wrapper running fine on 1703
Update 1703 >> 1709
RDP broken
RECOPIED the rfxvmt.dll to System32 as the Upgrade removed it
Reboot Win-10  -  RDP Wrapper still broken
re-ran the install.bat   I just clicked the bat file
Everything working again

It has worked One time in a row for me

**Comment by sickingajay** [comment]
![rdpwrap](https://user-images.githubusercontent.com/28782626/34068822-7c6af922-e268-11e7-979e-f7a1b8bb40e6.PNG)
was working fine until yesterday 
but seems not to work no more

**Comment by Steve-1029** [comment]
Sickingajay-
Did your computer do a forced Upgrade to ver 1709 ?

Use to work on 1703 abut now "Upgraded" ?

**Comment by sickingajay** [comment]
@Steve-1029 
No, as far as I remember.

**Comment by Steve-1029** [comment]
Check these things
do  start >> run >> winver      this will tell you what release of win-10
then check c:\windows\systme32  to see if the file rfxvmt.dll is in there

On my computer when Win-10 was on version 1703 all working with the RDP wrap 1.6.1
and the rfxvmt.dll  fix
but when windows upgraded itself to 1709 the dll file was removed

**Comment by sickingajay** [comment]
@Steve-1029 
windows 1709
rfxvmt.dll was removed 
placed it again in system32
but still gives the same error
can't seem to find any fix to this recent upgrade

**Comment by Steve-1029** [comment]
Ok
So after copying the dell reboot your machine

Then hit the install.bat script

Then the system went all Green for me

**Comment by dvdp** [comment]
Got it running again too on Win10 Home x64 after the Creator's Update - 1709:
Copied 32 bit dll to system32 and 64 bit dll to SysWOW64, ran uninstall.bat, next install.bat, reboot and all green lights. I guess uninstall/install isn't really needed but it didn't hurt ...

**Comment by sickingajay** [comment]
Tried almost everything said by everyone here
but it still says 'not listening'
Still looking for an answer 
thanks
![capture](https://user-images.githubusercontent.com/28782626/34111672-6d14f346-e430-11e7-85a3-2871ddd50b42.PNG)

**Comment by w17** [comment]
>Copied 32 bit dll to system32 and 64 bit dll to SysWOW64...
64 bit to system32 and 32 bit to SysWOW64 ?

**Comment by sickingajay** [comment]
can someone provide the link for 32 bit and 64 bit dll files

**Comment by explorerhk** [comment]
Have you read the whole thread?

**Comment by Idontgetit** [comment]
I'm having the same problem as @sickingajay .  Running 1709 build 16299.125 and tried multiple times all steps listed (including re-downloading all files) and still no luck.  Checked firewall also.  Any other suggestions?

**Comment by SpirravipDK** [comment]
I also have this issue...

Have tryed all thread hints and tips with no luck, 

Stuck on the NOT LISTENING and have win 10 home (1709 build 16299.125) (x64) (DK language)

Have tryed copying and replacing dll files in all mentioned folders, also rebootet after each try. have tryed with both 32 & 64 bit versions of dll's and mixed them (just to try)
Have used admin rights on all cmds. Have tryed giving ALL users all rights to dll's and tryed disabling the firewall completely.

Result = NOT LISTENING and no response for RDP connections. Even tryed to change the RDP port = still no luck...

Hope someone has a trick for this :)

/ Merry christmas everyone ....

**Comment by raisen** [comment]
I'm also not being able to get this to work with build 1709 16299.125 (x64).

I tried copying both the 32-bit and 64-bit DLL to their respective folders (I've also tried to copy only the 64-bit DLL), rebooted many times, uninstalled and installed rdpwrap multiple times.

**Comment by raisen** [comment]
To everyone with problems with build 1709 16299.125, I was just able to solve by following the instructions from sDunkan: https://github.com/stascorp/rdpwrap/issues/194#issuecomment-339586541 . 

Opening command prompt as administrator seemed to be what I was missing. Hopefully, it helps folks on this issue with the same problem.

**Comment by sickingajay** [comment]
So after restarting several time everything turned green
Just copy 64-bit dll file to system32
But the only thing that's not working is
whatever I do after taking RDP on my friends pc 
He's able to see everything
unchecked the single user as well but that doesn't seem to do anything
Anyone facing any similar issues?

**Comment by SpirravipDK** [comment]
![image](https://user-images.githubusercontent.com/34681739/34178587-0d14b9ee-e508-11e7-88c9-a90efe30a481.png)
@raisen , had tryed that already... and I have now tryed again... It still dosen't work... 

I have no clue of what to try next.... Everything mentioned in this tread have been tryed with/without admininstrator mode cmd, with/without reboots between each step, withboth x86 & x64 dll files.
With firewall on/disabled.... I'm almost giving up.....

**Comment by raisen** [comment]
@SpirravipDK I was almost giving up too as I was getting nowhere as well.

I used this link to download the DLL:

https://github.com/stascorp/rdpwrap/files/1236856/rfxvmt.zip

I had previously downloaded the DLL from this link: https://github.com/stascorp/rdpwrap/files/1438515/rfxvmt_x64_Build1709.zip

and it didn't work - maybe it's the reason for you too? I don't know...

I didn't have to reboot after I followed sDunkan's instructions.
